flutter/lib/resources/result_manager.dart

```diff
@@ -3,58 +3,31 @@ import 'dart:io';
 
 import 'package:mlperfbench/benchmark/benchmark.dart';
 import 'package:mlperfbench/benchmark/benchmark_result.dart';
-import 'package:mlperfbench/protos/backend_setting.pb.dart' as pb;
 
 class _BriefResult {
-  static const _idTag = 'benchmark_id';
-  static const _scoreTag = 'score';
-  static const _accuracyTag = 'accuracy';
-  static const _shardsNumTag = 'shards_num';
-  static const _batchSizeTag = 'batch_size';
-  static const _backendDescriptionTag = 'backend_description';
+  static const _tagId = 'benchmark_id';
+  static const _tagPerformance = 'performance';
+  static const _tagAccuracy = 'accuracy';
 
   final String id;
-  final double? score;
-  final String? accuracy;
-  final int shardsNum;
-  final int batchSize;
-  final String backendDescription;
+  final BenchmarkResult? performance;
+  final BenchmarkResult? accuracy;
 
   _BriefResult(
-      {required this.id,
-      required this.score,
-      required this.accuracy,
-      required this.shardsNum,
-      required this.batchSize,
-      required this.backendDescription});
-
-  Map<String, dynamic> toJsonMap() {
-    return {
-      _idTag: id,
-      _scoreTag: score,
-      _accuracyTag: accuracy,
-      _shardsNumTag: shardsNum,
-      _batchSizeTag: batchSize,
-      _backendDescriptionTag: backendDescription,
-    };
-  }
-
-  static _BriefResult fromJsonMap(Map<String, dynamic> jsonMap) {
-    final id = jsonMap[_idTag] as String;
-    final score = jsonMap[_scoreTag] as double?;
-    final accuracy = jsonMap[_accuracyTag] as String?;
-    final shardsNum = jsonMap[_shardsNumTag] as int? ?? 0;
-    final batchSize = jsonMap[_batchSizeTag] as int? ?? 0;
-    final backendDescription = jsonMap[_backendDescriptionTag] as String;
-
-    return _BriefResult(
-        id: id,
-        score: score,
-        accuracy: accuracy,
-        shardsNum: shardsNum,
-        batchSize: batchSize,
-        backendDescription: backendDescription);
-  }
+      {required this.id, required this.performance, required this.accuracy});
+
+  Map<String, dynamic> toJson() => {
```

Comment:
The name of the method is `toJson()` but return type is a `Map`. Maybe we can use `typedef Json = Map<String, dynamic>` here. The method signature would be then `Json toJson()`. And we can replace all `Map<String, dynamic>` in this file with `Json`

Reply:
This is the standard naming convention for Flutter.
https://docs.flutter.dev/development/data-and-backend/json#serializing-json-inside-model-classes
I don't think using `typedef Json` would improve code.
Json is `String` of certain format.
`Map<String, dynamic>` is not json, it's one of the types that are directly encodable using standard json tools in flutter.

Maybe we could name it `JsonMap` but I doubt it would really improve readability: everyone who uses flutter [theoretically] knows standard serializable types from flutter documentation, but only people working with `result_manager.dart` will know what `JsonMap` means.

Reply:
The example code is also for beginner, so they will keep it simple and don't want to explain another concept like type alias.
I think using `typedef` in this case make the code easier to read. But it doesn't impact the function of the app, so feel free to accept or ignore this change request.